src/hit_cylinder.c

```diff
@@ -0,0 +1,97 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   hit_cylinder.c                                     :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: u413q <u413q@student.42.fr>                +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/11/10 16:16:38 by sqiu              #+#    #+#             */
+/*   Updated: 2023/11/14 17:37:50 by u413q            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "hit_manager.h"
+
+bool	ft_hit_cylinder(t_cylinder cy, t_ray ray, t_hitrecord *rec, \
+	t_interval ray_d)
+{
+	float		potential_hits[4];
+
+	if (!ft_calculate_pot_hits(cy, ray, ray_d, potential_hits))
+		return (false);
+	ft_identify_hits(cy, ray, potential_hits, rec);
+	if (rec->d < EPSILON)
+		return (false);
+	rec->point = ft_ray(ray, rec->d);
+	rec->normal = ft_vec3_norm(ft_cylinder_normal(*rec, ray, cy));
+	rec->colour = cy.colour;
+	return (true);
+}
+
+bool	ft_calculate_pot_hits(t_cylinder cy, t_ray ray, t_interval ray_d, \
+	float potential_hits[4])
+{
+	t_equation	eq;
+	t_vec3		cap1_ray;
+	float		d3;
+	float		d4;
```

Comment:
nit: cap_hit_top and cap_hit_bot instead of d3 and d4?

Reply:
we dont know which is top and which bottom as the cylinder orientation can be anything. Also the first two solutions are named d1 and d2, therefore I'd keep it consistent

Reply:
I guess we could argue that till the end of days as each perspective has its merit. To put an end to it and as a my purpose in life is to make you happy, I renamed the two additional solutions cap_hit1 and 2

Reply:
Noice!